language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -642,7 +652,14 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
             }
 
             // TODO: Handle type actuals for generics
-            Bytecode::Exists(_, _) => {
+            Bytecode::Exists(idx, _) => {
+                let struct_definition = self.module.struct_def_at(*idx);
+                if !StructDefinitionView::new(self.module, struct_definition).is_resource() {
+                    return Err(VMStaticViolation::ExistsNoResourceError(offset));
+                } else if !state.global(*idx).is_empty() {
```

Comment:
nit, I don't like `else` in these sort of "guard"-like if statements. Personal preference though
(for now its probably best to stay in the same style of the rest of this file)